include/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -106,6 +106,9 @@ MEMORY
     SRAM1                 (rw) : ORIGIN = RAM1_ADDR, LENGTH = RAM1_SIZE
     SRAM2                 (rw) : ORIGIN = RAM2_ADDR, LENGTH = RAM2_SIZE
 #endif
+#ifdef CONFIG_ETH_STM32_HAL_USE_SRAM3
```

Comment:
The MEMORY description here already contains several other SoC-specific sections so having one more does not do any additional harm.

I would like to see this cleaned up in the future, and keep all SoC specific section in the corresponding SoC definitions.

Reply:
However, could we at least align with the more modern concept of doing this via Device Tree and not Kconfig?

Reply:
Done.